src/Vector3.h

```diff
@@ -393,11 +393,9 @@ class VectorHasher
 	{
 		// Guaranteed to have no hash collisions for any 128x128x128 area
 		size_t Hash = 0;
-		Hash ^= static_cast<size_t>(a_Vector.x);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.y);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.z);
+		Hash ^= a_Vector.x + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.y + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.z + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
```

Comment:
This hash function, [obtained here](http://stackoverflow.com/questions/20511347/a-good-hash-function-for-a-vector), was provided to easily change between maps and unordered_maps with regards to chunk storage, especially since there is a lot of theorising about the speed of the latter.

The previous hash function was providing outputs which caused an unordered_map to place all entries in around 10%? of buckets, which degraded performance.

Since you're all talking about the hash function, I assume the rest of my code is perfect? ;)